demos/projects/ESPRESSIF/aziotkit/components/azure-iot-kit-sensors/src/sensor_manager.c

```diff
@@ -0,0 +1,246 @@
+#define I2C_MASTER_FREQ_HZ 100000   /*!< I2C master clock frequency */
+
+#define BUTTON_IO_NUM  0
+#define BUTTON_ACTIVE_LEVEL   0
+
+#define SHAKE_THRESHOLD 2
+
+static i2c_bus_handle_t i2c_bus = NULL;
+static hts221_handle_t hts221 = NULL;
+static bh1750_handle_t bh1750 = NULL;
+static fbm320_handle_t fbm320 = NULL;
+static mag3110_handle_t mag3110 = NULL;
+static mpu6050_handle_t mpu6050 = NULL;
+static ssd1306_handle_t oled = NULL;
+static float range_per_digit = 0;
+
+/**
+ * @brief i2c master initialization
+ */
+static void i2c_master_init()
+{
+    int i2c_master_port = I2C_NUM_0;
+    i2c_config_t conf;
+    conf.mode = I2C_MODE_MASTER;
+    conf.sda_io_num = I2C_MASTER_SDA_IO;
+    conf.sda_pullup_en = GPIO_PULLUP_ENABLE;
+    conf.scl_io_num = I2C_MASTER_SCL_IO;
+    conf.scl_pullup_en = GPIO_PULLUP_ENABLE;
+    conf.clk_flags = 0;
+    conf.master.clk_speed = I2C_MASTER_FREQ_HZ;
+
+    i2c_bus = iot_i2c_bus_create(i2c_master_port, &conf);
+}
+
+void init_humiture_sensor()
+{
+    hts221 = iot_hts221_create(i2c_bus, HTS221_I2C_ADDRESS);
+
+    hts221_config_t hts221_config;
+    hts221_config.avg_h = HTS221_AVGH_32;
+    hts221_config.avg_t = HTS221_AVGT_16;
+    hts221_config.odr = HTS221_ODR_1HZ;
+    hts221_config.bdu_status = HTS221_DISABLE;
+    hts221_config.heater_status = HTS221_DISABLE;
+    iot_hts221_set_config(hts221, &hts221_config);
+    
+    iot_hts221_set_activate(hts221);
+}
+
+void init_ambient_light_sensor()
+{
+    bh1750 = iot_bh1750_create(i2c_bus, BH1750_I2C_ADDRESS);
+    bh1750_cmd_measure_t cmd_measure = BH1750_CONTINUE_4LX_RES;
+    iot_bh1750_power_on(bh1750);
+    iot_bh1750_set_measure_mode(bh1750, cmd_measure);
+}
+
+void init_motion_sensor()
+{
+    uint8_t range;
+    mpu6050 = iot_mpu6050_create(i2c_bus, MPU6050_I2C_ADDRESS);
+
+    mpu6050_init(mpu6050);
+    mpu6050_get_full_scale_accel_range(mpu6050, &range);
+
+    switch (range)
+    {
+    case 0:
+        range_per_digit = .000061f;
+        break;
+    case 1:
+        range_per_digit = .000122f;
+        break;
+    case 2:
+        range_per_digit = .000244f;
+        break;
+    case 3:
+        range_per_digit = .0004882f;
+        break;
+    default:
+        range_per_digit = .000061f;
+        break;
+    }
+}
+
+void init_barometer_sensor()
+{
+    fbm320 = iot_fbm320_create(i2c_bus, FBM320_I2C_ADDRESS);
+    fbm320_init(fbm320);
+}
+
+void init_magnetometer_sensor()
+{
+    mag3110 = iot_mag3110_create(i2c_bus, MAG3110_I2C_ADDRESS);
+    mag3110_start(mag3110);
+}
+
+void init_oled()
+{
+    oled = iot_ssd1306_create(i2c_bus, SSD1306_I2C_ADDRESS);
+    oled_init(oled);
+}
+
+void initialize_sensors()
+{
+    i2c_master_init();
+    init_humiture_sensor();
+    init_ambient_light_sensor();
+    init_motion_sensor();
+    init_barometer_sensor();
+    init_magnetometer_sensor();
+    init_oled();
+    initialize_leds();
+}
+
+void oled_show_message( const uint8_t * pucMessage, uint32_t ulMessageLength )

```

Comment:
>pucMessage [](http://example.com/codeflow?start=40&length=10)

we should refrain using type in name of variable ... this is something followed in freertos not in this file.

Reply:
Lets keep the consistency of this file .... space between parathesesis and next character is something done in freertos not in this file..

Reply:
Done.